File: feature_tests/java/somelib/src/test/java/dev/diplomattest/somelib/Float64VecTest.java
package dev.diplomattest.somelib;

import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.*;

class Float64VecTest {
    @Test
    void testFloat64Vec() {
        long[] lArray = {1, 2, 3, 4};

        double[] dArray = {1, 2, 3, 4};
        var float64Vec = Float64Vec.newIsize(lArray);
        var outArray = float64Vec.asSlice();
        assertEquals(dArray.length, outArray.length);
        for (var i = 0; i < dArray.length; i++) {
            assertEquals(dArray[i], outArray[i]);
        }

        var newVec = Float64Vec.new_(dArray);
        var newArray = newVec.asSlice();
        assertEquals(dArray.length, newArray.length);
        for (var i = 0; i < dArray.length; i++) {
            assertEquals(dArray[i], newArray[i]);
        }
    }

}

File: feature_tests/kotlin/somelib/src/test/kotlin/dev/diplomattest/somelib/Float64VecTest.kt
package dev.diplomattest.somelib

import net.jqwik.api.ForAll
import net.jqwik.api.Property
import net.jqwik.api.constraints.NotEmpty
import net.jqwik.api.constraints.Size
import net.jqwik.api.constraints.UniqueElements
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class Float64VecTest {
    @Test
    fun testFloat64Vec() {
        val doubleList = listOf(1.0, 2.0, 3.0, 4.0)
        val doubleArray = doubleList.toDoubleArray()
        val float64Array = Float64Vec.newFromOwned(doubleArray)
        val float64ArrayStr = float64Array.toString()
        assertEquals(float64ArrayStr, doubleList.toString())
    }

}

class Float64PropTest {
    @Property(tries = 1000)
    fun `should generate Lists of Doubles`(
        @ForAll @NotEmpty @Size(value = 1000) numbers: @UniqueElements List<Double>
    ) {
        val doubleArray = numbers.toDoubleArray()
        val float64Array = Float64Vec.newFromOwned(doubleArray)
        val float64List = float64Array.borrow().toList()
        assertEquals(float64List, numbers)
    }

}
